src/main.cpp

```diff
@@ -3182,20 +3182,21 @@ bool CBlock::ConnectBlock(CTxDB& txdb, CBlockIndex* pindex, bool fJustCheck, boo
                         pindex->nHeight,CoinToDouble(nTotalCoinstake),CoinToDouble(nTxValueOut));
                 }
 
-                // Verify no recipients exist after coinstake (Recipients start at output position 3 (0=Coinstake flag, 1=coinstake amount, 2=splitstake amount)
-                if (bIsDPOR && pindex->nHeight > nGrandfather)
+                if (pindex->nVersion >= 10)
                 {
+                    if (tx.vout.size() > 8)
+                        return DoS(100,error("Too many coinstake outputs"));
+                }
+                else if (bIsDPOR && pindex->nHeight > nGrandfather && pindex->nVersion < 10)
+                {
+                    // Old rules, does not make sense
+                    // Verify no recipients exist after coinstake (Recipients start at output position 3 (0=Coinstake flag, 1=coinstake amount, 2=splitstake amount)
```

Comment:
Do we verify the total sum elsewhere?

Reply:
Yes. SplitCoinStakeOutput will not allow more than 8 for V10. In #1244 I limited it to 2 in GetNumberOfStakeOutputs, which is called by SplitCoinStakeOutput, but the limit more properly belongs in SplitCoinStakeOutput.

Reply:
Hopefully it is checked somewhere at all. idk. We need to find it. I am pretty sure it is.

Reply:
I think we are ok... here is main.cpp:3167

            if (tx.IsCoinStake())
            {
                nStakeReward = nTxValueOut - nTxValueIn;
                if (tx.vout.size() > 3 && pindex->nHeight > nGrandfather) bIsDPOR = true;
                // ResearchAge: Verify vouts cannot contain any other payments except coinstake: PASS (GetValueOut returns the sum of all spent coins in the coinstake)
                if (IsResearchAgeEnabled(pindex->nHeight) && fDebug10)
                {
                    int64_t nTotalCoinstake = 0;
                    for (unsigned int i = 0; i < tx.vout.size(); i++)
                    {
                        nTotalCoinstake += tx.vout[i].nValue;
                    }
                    if (fDebug10)   LogPrintf(" nHeight %d; nTCS %f; nTxValueOut %f",
                        pindex->nHeight,CoinToDouble(nTotalCoinstake),CoinToDouble(nTxValueOut));
                }

and then main.cpp:3222

    double dStakeReward = CoinToDouble(nStakeReward+nFees) - DPOR_Paid; //DPOR Recipients checked above already
    double dStakeRewardWithoutFees = CoinToDouble(nStakeReward) - DPOR_Paid;

and finally main.cpp:3329

                if ((bb.ResearchSubsidy + bb.InterestSubsidy + dDrift) < dStakeRewardWithoutFees)
                {
                        return DoS(20, error("ConnectBlock[] : Researchers Interest %f + Research %f + TimeDrift %f and total Mint %f, [StakeReward] <> %f, with Out_Interest %f, OUT_POR %f, Fees %f, DPOR %f  for CPID %s does not match calculated research subsidy",
                            (double)bb.InterestSubsidy,(double)bb.ResearchSubsidy,dDrift,CoinToDouble(mint),dStakeRewardWithoutFees,
                            (double)OUT_INTEREST,(double)OUT_POR,CoinToDouble(nFees),(double)DPOR_Paid,bb.cpid.c_str()));

                }

Note that since the construction of nStakeReward is from the sum of coinstake txn inputs - sum of coinstake txn outputs, it is already doing what I suggested, and is insensitive to the number of outputs, only the value.

Reply:
Duh... I found the same stuff you posted below... anyway we are good...

Reply:
line 3172 will make bIsDPOR = true if over 3 in vout size. i experienced this on testnet causing it to say i was being paid to much when trying to split my interest. u need to change it to 
`if (tx.vout.size() > 3 && pindex->nHeight > nGrandfather && pindex->nResearchSubsidy > 0) bIsDPOR = true;`

Reply:
or change it to different approach either or it assumed my interest block was POR there

Reply:
also note oddly it seems we never reach a vout.size() > 3 is only settin bIsDPOR = true when we have 3 outputs. thou we dont seem to do that in POR/POS its usually 2 so that area needs some changes beyond what i've mentioned.

Reply:
possible removal of it would suffice @denravonska